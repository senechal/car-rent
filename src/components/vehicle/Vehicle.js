import PropTypes from 'prop-types';
import { Icons } from '@reusable';
import {
  Link
} from 'react-router-dom';
import {
  VENDOR_LOGOS,
} from './constants'
const {
  AcIcon,
  DoorIcon,
  FuelIcon,
  AutomaticGear,
  ManualGear,
  SeatIcon,
  BagIcon,
} = Icons;


const formatPrice = (price, code) => {
  return new Intl.NumberFormat('en-UK',
    { style: 'currency', currency: code }
  ).format(price);
};

const Vehicle = (props) => {

  const {
    id,
    status,
    hasAirConditioning,
    transmissionType,
    fuelType,
    vendor,
    seats,
    baggageQuantity,
    doors,
    makeModel,
    imgUrl,
    price,
    currencyCode,
  } = props;

  const [name, similar] = makeModel.split(' or ');

  return (
    <section className="rac-vehicle">
      <div className="rac-vehicle-content">
        <div className="rac-vehicle-img">
          <Link to={`/details/${id}`}><img src={imgUrl} alt={makeModel} /></Link>
        </div>
        <div className="rac-vehicle-info">
          <div className="rac-vehicle-make-model">
            <Link to={`/details/${id}`}><strong>{name}</strong> or {similar} ({status})</Link>
          </div>
          <div className="rac-vehicle-features">
            <span><SeatIcon /> {seats} seats </span>
            <span><DoorIcon /> {doors} doors </span>
          </div>
          <div className="rac-vehicle-features">
            <span><BagIcon /> {baggageQuantity} bags </span>
            <span><AcIcon /> { hasAirConditioning ? 'Air Conditioning' : '-' } </span>
          </div>
          <div className="rac-vehicle-features">
            <span><FuelIcon /> {fuelType} </span>
            {
              transmissionType === 'Automatic'
                ? <span><AutomaticGear /> Automatic Transmission</span>
                : <span><ManualGear /> Manual Transmission</span>
            }
          </div>
        </div>
        <div className="rac-vehicle-pricing">
          {formatPrice(price, currencyCode)}
        </div>
      </div>
      <div className="rac-vehicle-vendor">
        <strong>Vendor: </strong>
        <img src={VENDOR_LOGOS[vendor]} alt={vendor} />
      </div>
    </section>
  )
};

Vehicle.propTypes = {
  id: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  hasAirConditioning: PropTypes.bool.isRequired,
  transmissionType: PropTypes.string.isRequired,
  fuelType: PropTypes.string.isRequired,
  vendor: PropTypes.string.isRequired,
  seats: PropTypes.string.isRequired,
  baggageQuantity: PropTypes.string.isRequired,
  doors: PropTypes.string.isRequired,
  makeModel: PropTypes.string.isRequired,
  imgUrl: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  currencyCode: PropTypes.string.isRequired,
};

export default Vehicle;