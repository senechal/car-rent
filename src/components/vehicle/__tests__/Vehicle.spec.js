import { render } from '@testing-library/react';
import Vehicle from '../Vehicle';

const mockVehicle = {
  baggageQuantity: "2",
  currencyCode: "CAD",
  doors: "2",
  fuelType: "Petrol",
  hasAirConditioning: true,
  id: "HERTZ_1",
  imgUrl: "https://ctimg-fleet.cartrawler.com/kia/rio/primary.png?auto=format&w=600",
  makeModel: "Kia Rio or similar",
  price: "731.47",
  seats: "5",
  status: "Available",
  transmissionType: "Automatic",
  vendor: "HERTZ",
};

jest.mock('react-router-dom', () => ({
  Link: ({children, to}) => <a href={to}>{children}</a>,
}));

describe('Testing Vehicle', () => {
  it('should render correctly', () => {
    const { container } = render(<Vehicle {...mockVehicle}/>);
    expect(container).toMatchSnapshot();
  });
});
