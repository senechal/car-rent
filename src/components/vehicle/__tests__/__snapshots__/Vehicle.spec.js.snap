// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Vehicle should render correctly 1`] = `
<div>
  <section
    class="rac-vehicle"
  >
    <div
      class="rac-vehicle-content"
    >
      <div
        class="rac-vehicle-img"
      >
        <a
          href="/details/HERTZ_1"
        >
          <img
            alt="Kia Rio or similar"
            src="https://ctimg-fleet.cartrawler.com/kia/rio/primary.png?auto=format&w=600"
          />
        </a>
      </div>
      <div
        class="rac-vehicle-info"
      >
        <div
          class="rac-vehicle-make-model"
        >
          <a
            href="/details/HERTZ_1"
          >
            <strong>
              Kia Rio
            </strong>
             or 
            similar
             (
            Available
            )
          </a>
        </div>
        <div
          class="rac-vehicle-features"
        >
          <span>
            <span
              class="anticon rac-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                focusable="false"
                height="1em"
                width="1em"
              >
                seats.svg
              </svg>
            </span>
             
            5
             seats 
          </span>
          <span>
            <span
              class="anticon rac-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                focusable="false"
                height="1em"
                width="1em"
              >
                doors.svg
              </svg>
            </span>
             
            2
             doors 
          </span>
        </div>
        <div
          class="rac-vehicle-features"
        >
          <span>
            <span
              class="anticon rac-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                focusable="false"
                height="1em"
                width="1em"
              >
                bag.svg
              </svg>
            </span>
             
            2
             bags 
          </span>
          <span>
            <span
              class="anticon rac-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                focusable="false"
                height="1em"
                width="1em"
              >
                ac.svg
              </svg>
            </span>
             
            Air Conditioning
             
          </span>
        </div>
        <div
          class="rac-vehicle-features"
        >
          <span>
            <span
              class="anticon rac-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                focusable="false"
                height="1em"
                width="1em"
              >
                fuel.svg
              </svg>
            </span>
             
            Petrol
             
          </span>
          <span>
            <span
              class="anticon rac-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                focusable="false"
                height="1em"
                width="1em"
              >
                automatic.svg
              </svg>
            </span>
             Automatic Transmission
          </span>
        </div>
      </div>
      <div
        class="rac-vehicle-pricing"
      >
        CA$731.47
      </div>
    </div>
    <div
      class="rac-vehicle-vendor"
    >
      <strong>
        Vendor: 
      </strong>
      <img
        alt="HERTZ"
        src="hertz_logo.gif"
      />
    </div>
  </section>
</div>
`;
